add infra to cpou upgrade

qiliRedHat opened the pull request:
https://issues.redhat.com/browse/OCPQE-27104

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
First 24 nodes [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58965/rehearse-58965-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes/1859136122773311488) failed for "Process did not finish before 8h0m0s timeout"
Compared to a previous successful [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58096/rehearse-58096-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-cpou-aws-416to418-24nodes/1853658593711099904) without infra, in the failed job the  infra related steps took extra ~ 30 minutes.
```
INFO[2024-11-20T08:08:05Z] Running step loaded-upgrade-416to418-24nodes-openshift-qe-workers-infra-workload. 
INFO[2024-11-20T08:31:02Z] Step loaded-upgrade-416to418-24nodes-openshift-qe-workers-infra-workload succeeded after 22m57s. 
INFO[2024-11-20T08:31:02Z] Running step loaded-upgrade-416to418-24nodes-openshift-qe-move-pods-infra. 
INFO[2024-11-20T08:37:30Z] Step loaded-upgrade-416to418-24nodes-openshift-qe-move-pods-infra succeeded after 6m27s. 
```
With out infra
```
INFO[2024-11-05T08:47:14Z] Step loaded-upgrade-cpou-aws-416to418-24nodes-cucushift-chainupgrade-toimage succeeded after 2h58m3s. 
```
With infra
```
INFO[2024-11-20T11:08:47Z] Step loaded-upgrade-416to418-24nodes-cucushift-chainupgrade-toimage failed after 2h10m48s. 
INFO[2024-11-20T11:14:13Z] Running step loaded-upgrade-416to418-24nodes-gather-extra. 
{"component":"entrypoint","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:169","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 8h0m0s timeout","severity":"error","time":"2024-11-20T15:26:22Z"}
```
I'll increase the job timeout to 10h considering 120 worker nodes + ipsec may make longer time.

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
The failure is not related to the TIMEOUT. The Cluster Operator machine-config is degraded because ' MachineConfigPool infra has not progressed to latest configuration'. 
```
= Update Health =
Message: Cluster Operator machine-config is degraded (RequiredPoolsFailed)
  Since:       58m9s
  Level:       Error
  Impact:      API Availability
  Reference:   https://github.com/openshift/runbooks/blob/master/alerts/cluster-monitoring-operator/ClusterOperatorDegraded.md
  Resources:
    clusteroperators.config.openshift.io: machine-config
  Description: Unable to apply 4.17.0-0.nightly-2024-11-21-052346: error during syncRequiredMachineConfigPools: [context deadline exceeded, MachineConfigPool infra has not progressed to latest configuration: controller version mismatch for rendered-infra-6c171d9d397c09f3d4b0b81d46df2c05 expected 39e1cd3c3b04229c48988be1fb7f99b95856aff3 has 4bb3364914c4dbcdfcc08b0914f402cdd38f014f: <unknown>, retrying]
Message: Cluster Version version is failing to proceed with the update (ClusterOperatorDegraded)
  Since:       3m58s
  Level:       Warning
  Impact:      Update Stalled
  Reference:   https://github.com/openshift/runbooks/blob/master/alerts/cluster-monitoring-operator/ClusterOperatorDegraded.md
  Resources:
    clusterversions.config.openshift.io: version
  Description: Cluster operator machine-config is degraded
Message: Outdated nodes in a paused pool 'infra' will not be updated
  Since:       -
  Level:       Warning
  Impact:      Update Stalled
  Reference:   https://docs.openshift.com/container-platform/latest/support/troubleshooting/troubleshooting-operator-issues.html#troubleshooting-disabling-autoreboot-mco_troubleshooting-operator-issues
  Resources:
    machineconfigpools.machineconfiguration.openshift.io: infra
  Description: Pool is paused, which stops all changes to the nodes in the pool, including updates. The nodes will not be updated until the pool is unpaused by the administrator.
Message: Outdated nodes in a paused pool 'worker' will not be updated
  Since:       -
  Level:       Warning
  Impact:      Update Stalled
  Reference:   https://docs.openshift.com/container-platform/latest/support/troubleshooting/troubleshooting-operator-issues.html#troubleshooting-disabling-autoreboot-mco_troubleshooting-operator-issues
  Resources:
    machineconfigpools.machineconfiguration.openshift.io: worker
  Description: Pool is paused, which stops all changes to the nodes in the pool, including updates. The nodes will not be updated until the pool is unpaused by the administrator.
```

qiliRedHat commented:
Opened a bug https://issues.redhat.com/browse/OCPBUGS-45045

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-24nodes

qiliRedHat commented:
120 nodes test [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58965/rehearse-58965-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes/1864181445954637824) failed 
```
worker rendered-worker-a7f84e5f437d132f60185decbabeca3a False True False 120 66 66 0 5h57m
{"component":"entrypoint","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:169","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 1h10m0s timeout","severity":"error","time":"2024-12-04T11:58:32Z"}
Checking worker pool status #10...
worker rendered-worker-a7f84e5f437d132f60185decbabeca3a False True False 120 74 74 0 6h2m
Checking worker pool status #11...
worker rendered-worker-a7f84e5f437d132f60185decbabeca3a False True False 120 85 85 0 6h7m
{"component":"entrypoint","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:267","func":"sigs.k8s.io/prow/pkg/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 10m0s grace period","severity":"error","time":"2024-12-04T12:08:32Z"}
```
85 in 120 finished. I'll extend the timeout: 1h10m to 2h.

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse ack

qiliRedHat commented:
@paigerube14 @liqcui  Please help to review.
120 node aws ovn cpou upgrade from 4.16 to 4.18: passed [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58965/rehearse-58965-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes/1864299989908328448)
For aws ipsec, there is a [bug](https://issues.redhat.com/browse/OCPBUGS-45102), so the test can not be run now. After the bug fix, I will enable cron jobs for aws ipsec cpou upgrade.

qiliRedHat commented:
@jiajliu Hi Jia, please help to review the change to 'extend the timeout of cucushift-upgrade-cpou-unpause-worker-mcp from 1h10m to 2h to support 120 worker nodes'. Thanks. You previous [pr](ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-ref.yaml) extended it to 1h10m.

jiajliu commented on ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.14-nightly-x86-cpou-loaded-upgrade-from-4.12.yaml:
```diff
@@ -28,59 +28,22 @@ resources:
       ITERATION_MULTIPLIER_ENV: "6"
+      MAX_UNAVAILABLE_WORKER: "3"
+      MCO_CONF_DAY2_CUSTOM_MCP: '[{"mcp_name": "infra"}]'
+      PAUSED_MCP_NAME: worker
```
I see only worker nodes paused, it seems not a control-plane-only update, right?

qiliRedHat replied:
@jiajliu Yes, the job runs control-plane-only update.

jiajliu replied:
so both worker and infra mcp should be paused?

qiliRedHat replied:
@jiajliu I only pause worker mcp with `PAUSED_MCP_NAME: worker`. 
For `MCO_CONF_DAY2_CUSTOM_MCP: '[{"mcp_name": "infra"}]'`, this is to overwrite the `expected_mcp` that only allow master and worker mcp by default [here](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/cpou/pause-worker-mcp/cucushift-upgrade-cpou-pause-worker-mcp-commands.sh#L45-L47)

qiliRedHat replied:
@jiajliu I only paused worker mcp `PAUSED_MCP_NAME: worker`
For `MCO_CONF_DAY2_CUSTOM_MCP: '[{"mcp_name": "infra"}]'`, it is to overwrite `expected_mcp` that only allow master and worker mcp by default https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/cpou/pause-worker-mcp/cucushift-upgrade-cpou-pause-worker-mcp-commands.sh#L45-L47. This step 'check all actual mcp, if any of them unknown then break the job.'

jiajliu replied:
> I only paused worker mcp PAUSED_MCP_NAME: worker

Hmm, for a control-plane-only update, all non-master mcp should be paused before upgrade. If your test requirement is to do cpou update, I guess both infra and worker are expected to be paused.

qiliRedHat replied:
@jiajliu From perfscale test team's point of view, either is ok for us. It will be good if I can get some guidance about what is recommended officially.
The current way I did (infra mcp not paused) is based on a pr [21175](https://github.com/openshift/release/pull/21175), and related Jira [OTA-448:](https://issues.redhat.com/browse/OTA-448) Add upgrade tests for a cluster with infra nodes. From the description of pr https://github.com/openshift/release/pull/21175, my understanding is the infra mcp is expected to be unpaused. 
I will start a thread between the pr owner and us to clarify it.

qiliRedHat replied:
https://redhat-internal.slack.com/archives/CH76YSYSC/p1733456153871259

jiajliu commented:
> please help to review the change to 'extend the timeout of cucushift-upgrade-cpou-unpause-worker-mcp from 1h10m to 2h to support 120 worker nodes'.

this part lgtm

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes

qiliRedHat commented:
@jiajliu I took your suggestion to pause infra mcp as well to make it a real 'cpou' upgrade. Added infra nodes, the unpause step needs more time to complete, I updated the timeout to 2h30m.
The new test [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58965/rehearse-58965-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.18-nightly-x86-cpou-loaded-upgrade-from-4.16-loaded-upgrade-416to418-120nodes/1865929391616299008) passed. 
Please help to review again. TIA!

qiliRedHat commented:
/pj-rehearse ack

jiajliu commented:
lgtm

jiajliu commented:
/lgtm

liqcui commented:
/lgtm
